### PR TITLE
fix: a user defined function with a syntax error prevented udfs with …

### PR DESCRIPTION
**Pull request by bogensberger**
…valid syntax

from being registered

### CODE REVIEW

File: sql/src/main/java/io/crate/operation/udf/UserDefinedFunctionService.java [hunk 1]
```diff
@@ -200,12 +201,22 @@ public void clusterChanged(ClusterChangedEvent event) {
...
-
-        functions.registerSchemaFunctionResolvers(functions.generateFunctionResolvers(udfFunctions));
+    static Map<FunctionIdent, FunctionImplementation> createFunctionImplementations(UserDefinedFunctionsMetaData metaData, ESLogger logger) {
+        Map<FunctionIdent, FunctionImplementation> udfFunctions = new HashMap<>();
+        for (UserDefinedFunctionMetaData function : metaData.functionsMetaData()) {
+            try {
+                FunctionImplementation impl = UserDefinedFunctionFactory.of(function);
+                udfFunctions.put(impl.info().ident(), impl);
+            } catch (javax.script.ScriptException e) {
+                logger.warn("Can't create user defined function", e);
```
Comment (seut) [inline sql/src/main/java/io/crate/operation/udf/UserDefinedFunctionService.java]: adding the name of the function to the message would be helpful

File: sql/src/test/java/io/crate/operation/udf/UserDefinedFunctionServiceTest.java [hunk 2]
```diff
@@ -102,4 +107,18 @@ public void testReplaceIsFalse() throws Exception {
...
         putFunction(UserDefinedFunctionsMetaData.of(same1), same2, false);
     }
+
+    @Test
+    public void testInvalidFunction() throws Exception {
+        UserDefinedFunctionMetaData invalid = new UserDefinedFunctionMetaData(
+            "invalid", ImmutableList.of(), ImmutableList.of(), DataTypes.INTEGER,
+            "javascript", "function invalid(){ this is not valid javascript code }"
+        );
+        UserDefinedFunctionsMetaData metaData = UserDefinedFunctionsMetaData.of(invalid, same1);
```
Comment (seut) [inline sql/src/test/java/io/crate/operation/udf/UserDefinedFunctionServiceTest.java]: pls make same1, same2 and different function definition static and rename to uppercase. they can be static and usage more obvious (I was first wondering why 1 registered function is expected by that test ;)

Comment (kovrus) [inline sql/src/test/java/io/crate/operation/udf/UserDefinedFunctionServiceTest.java]: @seut i think, @mfussenegger will be against making it static

Comment (seut) [inline sql/src/test/java/io/crate/operation/udf/UserDefinedFunctionServiceTest.java]: @bogensberger @kovrus ok got the point (randomized testrunner static leak assertion), leave it.

**Pull request merged**